src/core/container.rs

```diff
@@ -54,6 +91,34 @@ where
         container
     }
 
+    /// Sets the host given the docker client.
+    ///
+    /// to cache this value for frequent checks via `host`
+    fn set_host(docker_client: &'d D) -> String {
+        if let Some(host) = std::env::var("DOCKER_HOST").ok() {
+            let host_url = Url::parse(&host).expect("failed to parse url");
+            match host_url.scheme() {
+                "https" | "http" | "tcp" => {
+                    return host_url.host_str().unwrap().to_string();
+                }
+                _ => (),
+            }
+        }
+        if Path::new("/.dockerenv").exists() {
+            let container = docker_client.run(Alpine);
```

Comment:
TODO: this should be globally cached to avoid `docker run` for each new container

Reply:
This should be done with [`lazy_static`](https://docs.rs/lazy_static/1.2.0/lazy_static/) but I am unable to call the trait method `run` on a generic docker container inside of this macro.